### ISSUE TITLE
Problem with some Continental A2L

### ISSUE THREAD

**Issue by lndmnn** [Luncher91/A2LParser#38]
Hi there,

i do have some issues when trying to convert a Continental A2L to json.
I've created a paste to show the output. Of course i can supply you with that A2L, but i'd love to do that on a direct way.

Maybe you can take a look at it. INCA does take that A2L without any error.

https://pastebin.com/yYUEVzUg

**Comment by Luncher91** [comment]
It looks like the identifier of an element contains a round bracket ')', ')' which is - at least according to the 1.6 standard - not allowed.
To be sure and maybe fix some other issues it would be great to see the A2L file. Can you send it to me?
andreas aaatttt lord-luncher.de

I want the parser to be also usefull to identify wrong A2L file, so I would not just allow round brackets. But it should be fine if I add the round brackets plus adding a warning when round brackets are contained in the identifier.But first we should check if my guess is right and round brakcets are actually the problem :-D

**Comment by lndmnn** [comment]
Yeah, very strange.. Maybe we can fix that issue. I've send you an E-Mail with that A2L :)

**Comment by Luncher91** [comment]
So I checked your file and I was right: The A2L file contains identifiers with round brackets.
Not only that but some identifiers even contain '#'

```
IDENTIFIER : 
	[a-zA-Z_][a-zA-Z0-9_\-#]* 
	(('[' [a-zA-Z0-9_#]+ ']') | ('(' [a-zA-Z0-9_#]+ ')'))*
	([_.] [a-zA-Z0-9_\-#]* 
		(('[' [a-zA-Z0-9_#]+ ']') | ('(' [a-zA-Z0-9_#]+ ')'))*
	)*;
```

This is now the new identifier specification I came up with but I still need to implement a warning which ready in the file you sent me.
Next step is to write a check and a warning if there is a round bracket or a # in the identifier.

**Comment by lndmnn** [comment]
Great! Thanks for your time. Looking forward to a new release with a fix for that.

**Comment by Luncher91** [comment]
Fixed and released with release 2.6.3

**Comment by lndmnn** [comment]
Thank you!

Is that just a warning or does the parser ignore that?

Line 1694943@12: The identifier contains an invalid character '(',')','#': AIRM_Port_flap_power_stage_diagnosis_(Appl__Inc_)
